File: Observer/Go/observer/observerB.go
package observer

import "fmt"

type ObserverB struct {
	value string
}

func (ob ObserverB) Update(val string) {
	ob.value = val
	ob.Display()
}

func (ob ObserverB) Display() {
	fmt.Printf("observer B value is " + ob.value + "\n")
}

File: Observer/PHP/observerB.php
<?php
  class ObserverB implements Observer
  {
    private $_value;

    public function update($value)
    {
      $this->_value = $value;
      $this->display();
    }

    public function display() {
      echo "Observer B value is " . $this->_value . "\n";
    }
  }
?>
